File: api/src/main/java/com/fundmate/api/service/TransactionService.java
package com.fundmate.api.service;

import com.fundmate.api.dto.request.TransactionRequest;
import com.fundmate.api.dto.response.TransactionResponse;

import java.time.LocalDate;
import java.util.List;

public interface TransactionService {
    TransactionResponse createTransaction(TransactionRequest transactionRequest);
    List<TransactionResponse> getTransactionByAccountId(Long accountId);
    List<TransactionResponse> getTransactionsByDateRange(Long accountId, LocalDate startDate, LocalDate endDate);
    TransactionResponse getTransactionById(Long id);
    TransactionResponse updateTransaction(Long id, TransactionRequest transactionRequest);
    void deleteTransaction(Long id);
}

File: frontend/src/services/transaction.service.js
import api from "./api";

export const transactionService = {
  async getTransactionsByAccount(accountId) {
    const response = await api.get(`/transactions/account/${accountId}`);
    return response.data;
  },

  async getTransactionByDateRange(accountId, startDate, endDate) {
    const response = await api.get(
      `/transactions/accounts/${accountId}/date-range`,
      {
        params: { startDate, endDate },
      }
    );
    return response.data;
  },

  async createTransaction(transactionData) {
    const response = await api.post("/transactions", transactionData);
    return response.data;
  },

  async updateTransaction(id, transactionData) {
    const response = await api.put(`/transactions/${id}`, transactionData);
    return response.data;
  },

  async deleteTransaction(id) {
    const response = await api.delete(`/transactions/${id}`);
    return response.data;
  },
};
